src/main/java/com/exadel/discount/controller/CouponController.java

```diff
@@ -0,0 +1,68 @@
+package com.exadel.discount.controller;
+
+import com.exadel.discount.dto.coupon.CouponDto;
+import com.exadel.discount.dto.coupon.CreateCouponDto;
+import com.exadel.discount.service.CouponService;
+import io.swagger.annotations.ApiOperation;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Sort;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/coupons")
+public class CouponController {
+    private final CouponService couponService;
+
+    @GetMapping
+    @ApiOperation("Get list of all coupons with sorting")
+    /**  Get list of all users with sorting by params
+     sortDirection - ASC or DSC (unsorted - by default) ;
+     sortField - name of sorted field by (id - by default)- date **/
+    public List<CouponDto> getAllCoupons(@RequestParam(value = "sortDirection", defaultValue = "") String sortDirection,
+                                         @RequestParam(value = "sortField", defaultValue = "id") String sortField){
+        Sort sort = Sort.unsorted();
+        if(!sortDirection.equals("")) {
+            sort = Sort.by(Sort.Direction.valueOf(sortDirection.toUpperCase()), sortField);
+        }
```

Comment:
Maybe this part should be in service layer, not in controller

Reply:
Done